### ISSUE TITLE
Problems with using tune_model in pycaret 2

### ISSUE THREAD

**Issue by dxs8292nt** [pycaret/pycaret#1029]
I am running pycaret 2.2 on databricks, when I run the command

tuned_lr=tune_model(lr,n_iter=50,optimize='AUC',search_library='tune-sklearn',search_algorithm='hyperopt')

I get the following error when I submit this cell:

ModuleNotFoundError: No module named 'tune_sklearn._detect_xgboost'    

even if I change the search_algorithm parameter I get the same error

**Comment by pycaret** [comment]
@Yard1

**Comment by Yard1** [comment]
@dxs8292nt @pycaret This will be fixed in the next PyCaret release. For now, you can install the nightly version:

```
pip uninstall pycaret
pip install -U pycaret-nightly
```

**Comment by dxs8292nt** [comment]
I loaded pycaret-nightly and issued the same command : tuned_lr=tune_model(lr,n_iter=50,optimize='AUC',search_library='tune-sklearn',search_algorithm='hyperopt')

and got the following back

AttributeError: 'ConsoleBuffer' object has no attribute 'fileno'

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<command-2109610259897506> in <module>
----> 1 tuned_lr=tune_model(lr,n_iter=50,optimize='AUC',search_library='tune-sklearn',search_algorithm='hyperopt')

/local_disk0/pythonVirtualEnvDirs/virtualEnv-6f0688d6-4e9c-4c82-ad04-1ca16ec63adc/lib/python3.7/site-packages/pycaret/classification.py in tune_model(estimator, fold, round, n_iter, custom_grid, optimize, custom_scorer, search_library, search_algorithm, early_stopping, early_stopping_max_iters, choose_better, fit_kwargs, groups, return_tuner, verbose, tuner_verbose, **kwargs)
   1100         verbose=verbose,
   1101         tuner_verbose=tuner_verbose,
-> 1102         **kwargs,
   1103     )
   1104 

/local_disk0/pythonVirtualEnvDirs/virtualEnv-6f0688d6-4e9c-4c82-ad04-1ca16ec63adc/lib/python3.7/site-packages/pycaret/internal/tabular.py in tune_model_supervised(estimator, fold, round, n_iter, custom_grid, optimize, custom_scorer, search_library, search_algorithm, early_stopping, early_stopping_max_iters, choose_better, fit_kwargs, groups, return_tuner, verbose, tuner_verbose, display, **kwargs)
   4471                     model_grid.fit(X_train, y_train, groups=groups, **fit_kwargs)
   4472         else:
-> 4473             model_grid.fit(X_train, y_train, groups=groups, **fit_kwargs)
   4474         best_params = model_grid.best_params_
   4475         logger.info(f"best_params: {best_params}")

/local_disk0/pythonVirtualEnvDirs/virtualEnv-6f0688d6-4e9c-4c82-ad04-1ca16ec63adc/lib/python3.7/site-packages/tune_sklearn/tune_basesearch.py in fit(self, X, y, groups, **fit_params)
    650                         configure_logging=False,
    651                         ignore_reinit_error=True,
--> 652                         include_dashboard=False)
    653                 else:
    654                     ray.init(

/local_disk0/pythonVirtualEnvDirs/virtualEnv-6f0688d6-4e9c-4c82-ad04-1ca16ec63adc/lib/python3.7/site-packages/ray/worker.py in init(address, num_cpus, num_gpus, resources, object_store_memory, local_mode, ignore_reinit_error, include_dashboard, dashboard_host, dashboard_port, job_config, configure_logging, logging_level, logging_format, log_to_driver, _enable_object_reconstruction, _redis_max_memory, _plasma_directory, _node_ip_address, _driver_object_store_memory, _memory, _redis_password, _java_worker_options, _code_search_path, _temp_dir, _load_code_from_local, _lru_evict, _metrics_export_port, _system_config)
    771         driver_object_store_memory=_driver_object_store_memory,
    772         job_id=None,
--> 773         job_config=job_config)
    774 
    775     for hook in _post_init_hooks:

/local_disk0/pythonVirtualEnvDirs/virtualEnv-6f0688d6-4e9c-4c82-ad04-1ca16ec63adc/lib/python3.7/site-packages/ray/worker.py in connect(node, mode, log_to_driver, worker, driver_object_store_memory, job_id, job_config)
   1078     try:
   1079         if not faulthandler.is_enabled():
-> 1080             faulthandler.enable(all_threads=False)
   1081     except io.UnsupportedOperation:
   1082         pass  # ignore

AttributeError: 'ConsoleBuffer' object has no attribute 'fileno'

**Comment by Yard1** [comment]
Can you create a new virtual environment and try again?

**Comment by Yard1** [comment]
I cannot replicate this issue on Google Colab.

**Comment by mihird1010** [comment]
@dxs8292nt, you can either upgrade pycaret to nightly as @Yard1 pointed out or downgrade tune-sklearn while keeping the same pycaret. The error is because the tune_sklearn._detect_xgboost is renamed to tune_sklearn._detect_booster in the later versions of tune_sklearn.

Tl;dr:
`
$ pip uninstall tune-sklearn
$ pip install tune-sklearn==0.1.0
Run tune_model
`

**Comment by mihird1010** [comment]
@dxs8292nt, you can either upgrade pycaret to nightly as @Yard1 pointed out or downgrade tune-sklearn while keeping the same pycaret. The error is because the tune_sklearn._detect_xgboost is renamed to tune_sklearn._detect_booster in the later versions of tune_sklearn.

Tl;dr:
`$ pip uninstall tune-sklearn`
`$ pip install tune-sklearn==0.1.0`
Run tune_model
`

**Comment by Yard1** [comment]
tune-sklearn 0.2.0 compatibility will come in an update which will release this week.

**Comment by dxs8292nt** [comment]
That is great thanks for the update

**Comment by garyee** [comment]
I think this can be closed as the [release 2.3.0](https://github.com/pycaret/pycaret/releases) fixes it
